comment/api/views.py

```diff
@@ -45,7 +45,7 @@ class CommentDetail(generics.RetrieveUpdateDestroyAPIView):
     permission_classes = (permissions.IsAuthenticatedOrReadOnly, IsOwnerOrReadOnly)
 
 
-class CommentDetailForReaction(generics.RetrieveAPIView):
+class CommentDetailForReaction(generics.UpdateAPIView):
```

Comment:
There won't be any change on the comment itself, why would we need UpdateAPIView?

Reply:
yes but the `flag` object is part of the comment instance. so we are actually updating an instance here.

Reply:
`RetrieveAPIView` is anyway used for read-only situations.

Reply:
> `RetrieveAPIView` is anyway used for read-only situations.

This is exactly what is intended to be when updating one of the comment components (flag, reaction ...), I don't want the comment to be accidentally updated.

Reply:
what is the object under consideration for the view? it's a comment, a part of which is getting modified.

for what it's worth, the name of the view doesn't restrict anything, it just gives you some handlers for some cases. In our case, we are anyway overriding the `post` method and manually returning the deserialized instance.

Reply:
The view name is matter here, this will give the user hint that the comment itself should not be updated within this view.

Reply:
Are you insisting on using `UpdateAPIView`? If yes, please test the affected endpoints by passing the comment content in the body and if you get the expected result, we can then merge it.

Reply:
don't you think the current tests don't cover this?

Reply:
I don't think that unit tests cover 100% of the cases even if the coverage ratio is 100%. E.g. The unit test was covering create comment in the serializer but we found that `validate_email` method was not called as expected. Therefore integration test is needed for some cases.

Reply:
okay, i have added a test to demonstrate this.